### DISCUSSION TITLE
Number of runs too small when using ParameterSpace

### DISCUSSION THREAD

**Discussion by leotozzi88** [neurolib-dev/neurolib#198] [Q&A] [answered]
Dear Experts, 

I would like to run simulations exploring a parameter space for an ALN model. 

I have read your examples online and I have managed to write my model evaluation function and run the search as follows:

```
parameters=ParameterSpace({"Ke_gl":[0.1, 500.0], "sigma_ou":[0.2, 0.3]})
search=BoxSearch(aln, parameters, evaluate)
search.run()

```
When I run this command, I get: 

```
MainProcess root INFO     Number of processes: 8
MainProcess pypet.storageservice.HDF5StorageService INFO     I will use the hdf5 file `./data/hdf/exploration.hdf`.
MainProcess pypet.environment.Environment INFO     Environment initialized.
MainProcess root INFO     Number of parameter configurations: 2
MainProcess root INFO     BoxSearch: Environment initialized.

```
The simulations run, but I only get 2 runs of the simulation, whereas I would expect to have 4 (4 possible combinations of parameters). 

I might be missing something obvious, but how can I use ParameterSpace to explore all possible combinations of parameters? 

Thank you, 

Leonardo Tozzi

**Comment by jajcayn** [comment, marked as answer]
Dear @leotozzi88,

For simple grids with only 2 values per parameter, neurolib is trying to automatically infer what kind of space you want to explore and incorrectly assumes those are just bounds. This should work:
```python
...
parameters=ParameterSpace({"Ke_gl":[0.1, 500.0], "sigma_ou":[0.2, 0.3]}, kind="grid")
...
```

Thanks for using `neurolib`:)

N.

**Reply by leotozzi88** [reply]
Oh wow the library is too smart! :D 

Thank you! 

Leonardo Tozzi
